spec/dry/types/schema_spec.rb

```diff
@@ -315,4 +315,57 @@ def self.to_ary
       end
     end
   end
+
+  describe '#to_s' do
+    context 'empty schema' do
+      subject(:type) { Dry::Types['hash'].schema({}) }
+
+      it 'returns string representation of the type' do
+        expect(type.to_s).to eql('#<Dry::Types[Schema<keys={}>]>')
+      end
+    end
+
+    context 'strict schema' do
+      subject(:type) { Dry::Types['hash'].schema({}).strict }
+
+      it 'returns string representation of the type' do
+        expect(type.to_s).to eql('#<Dry::Types[Schema<keys={} strict>]>')
+      end
+    end
+
+    context 'key transformation' do
+      let(:key_transformation) { :to_sym.to_proc }
+
+      let(:transformation_name) { Dry::Types::FnContainer.register_name(key_transformation) }
+
+      subject(:type) { Dry::Types['hash'].schema({}).with_key_transform(key_transformation) }
+
+      it 'returns string representation of the type' do
+        expect(type.to_s).
+          to eql("#<Dry::Types[Schema<keys={} key_fn=#{ transformation_name }.call>]>")
```

Comment:
Is there a way we can specify the output explicitly here, without relying on `transformation_name.to_s` via this interpolation? Just reading through the specs as you suggested and this one made it harder for me to understand the final output.

Reply:
not really, its name is dynamic because functions/procs are registered globally in `Dry::Types::FnContainer`, it'll be something like `fn_70170547805360`

Reply:
@flash-gordon Ah, fair enough.